compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -373,6 +376,7 @@ public BType checkExpr(BLangExpression expr, SymbolEnv env, BType expType, Diagn
 
         expr.accept(this);
 
+        resultType = types.getConstraintFromReferenceType(resultType);
```

Comment:
These sort of usages of this method is a bit problematic since it leads to the same problem we had before: losing the type ref info. e.g., 
```ballerina
public type Err1 error;

public function main() {
    Err1 err;
    error e = <cursor>err;
}
```
At `<cursor>`, if you lookup the type of the `err` var ref, we get `error` as the type. But ideally, this should've been a type ref for `Err1`.

Reply:
This is also problematic for the `symbol()` method's usages. e.g.,
```ballerina
type Foo int;

function test() {
    Foo <cursor>f = 10;
}
```
If we take `f`'s symbol from the illustrated cursor position, and then get `f`'s type, what we get is the `int` type, not `Foo`.

Reply:
Addressed in 75553bb1688dd06b0e4e9a8d73908ccf7f815b3c